navigation/src/main/java/template/navigation/DynamicNavigationContainer.kt

```diff
@@ -0,0 +1,148 @@
+package template.navigation
+
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.material3.PermanentNavigationDrawer
+import androidx.compose.material3.Surface
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.tooling.preview.Devices
+import androidx.compose.ui.tooling.preview.Preview
+
+@Composable
+fun DynamicNavigationContainer(
+    navigationItems: List<NavigationTabDisplayModel>,
+    navigationType: NavigationType,
+    modifier: Modifier = Modifier,
+    appContent: @Composable () -> Unit,
+) {
+    when (navigationType) {
+        NavigationType.BOTTOM_NAVIGATION -> {
+            BottomNavigationContainer(navigationItems, modifier, appContent)
```

Comment:
One subtle state issue with the current setup:

Because `appContent` will be called in different places in the code depending on `navigationType`, any state internal to `appContent` (`remember` and `rememberSaveable`) will be lost when the `navigationType` changes. There's a strong expectation that this won't happen when calling a component with a slot: https://android.googlesource.com/platform/frameworks/support/+/androidx-main/compose/docs/compose-component-api-guidelines.md#lifecycle-expectations-for-slot-parameters

Since this component likely lives near the root of the UI hierarchy with most of the app's content inside it, this will cause severe state loss issues upon window resizing.

Two possible options would be to use a custom layout to only need to call `appContent` in one place, or use `movableContentOf` as an implementation detail of `DynamicNavigationContainer` to move the content around and preserve the lifecycle of `appContent`.

Reply:
Thanks a ton! Did not realize that was a potential concern. I'll read through these docs and make a change.

Reply:
I feel like `movableContentOf` is probably the most readable solution, as that would live at the entry to `DynamicNavigationContainer`, and then readers can follow that downstream into the sub components? Curious about your thoughts on that vs the other solution in the docs.

Reply:
Made a change, but I'll also try to use this in an actual sample scenario so I can test this out and make sure it behaves as expected.

Reply:
Tested briefly using a checkbox example like the docs and it looks good. Thanks again!